system/assets/hypha-preview-image-file-field.js

```diff
@@ -0,0 +1,64 @@
+$(function() {
+	$('img[data-preview-for]').each(function () {
+		// get preview
+		var $preview = $(this);
+
+		// get input name
+		var inputName = $preview.data('preview-for');
+
+		// get input
+		var $input = $('input[name="' + inputName + '"][type="file"]');
+
+		// add button to remove the image file
+		var addRemoveImageBtn = function () {
+			$('<span class="hyphaRemoveButton" />')
+				.on('click', function (e) {
+					e.preventDefault();
+
+					// remove delete button
+					$(this).remove();
+
+					updateSrc(null);
+				})
+				.insertAfter($preview);
+		}
+
+		var toggle = function (hasSrc) {
+			if (hasSrc) {
+				$input.hide();
+				$('label[for="' + inputName + '"]').attr('for', inputName + '_disabled');
+				addRemoveImageBtn();
+			} else {
+				$input.show();
+				$('label[for="' + inputName + '_disabled"]').attr('for', inputName);
+
+				// clear file value
+				$input.attr('value', null);
+			}
+		};
+
+		var updateSrc = function (src) {
+			let hasSrc = !!src;
+			if (hasSrc) {
+				$preview.attr('src', src);
+			} else {
+				$preview.removeAttr('src');
+			}
+
+			// update shadow file value
+			$('input[data-file-shadow-value-for="' + inputName + '"]').attr('value', src);
+
+			toggle(hasSrc);
+		}
+
+		var init = function () {
+			$input.on('change', function () {
+				updateSrc(window.URL.createObjectURL(this.files[0]));
+			});
+
+			toggle($preview.attr('src'));
```

Comment:
Why not merge `toggle` and `updateSrc` into one function?

I was also thinking hat if you use the value of the shadow field here, rather than the preview src, we could remove the preview handling on the PHP side, but that would prevent the preview from being resized (which might not be terrible, of course).

Reply:
> I was also thinking hat if you use the value of the shadow field here, rather than the preview src, we could remove the preview handling on the PHP side
This would not work because the shadow field value does not contain the correct path.

Reply:
Yeah, I guess. Still, merging `toggle` and `updateSrc` seems reasonable?

Reply:
merged